Frameworks/Foundation/NSURLSessionConfiguration.mm

```diff
@@ -24,56 +24,43 @@ @interface NSURLSessionConfiguration ()
 
 @implementation NSURLSessionConfiguration
 
-- (instancetype)_initWithSharedDefaults {
+/**
+ @Status Interoperable
+*/
+- (instancetype)init {
     if (self = [super init]) {
         _allowsCellularAccess = YES;
         _timeoutIntervalForRequest = 60;
-        _timeoutIntervalForResource = 7;
+        _timeoutIntervalForResource = 7 * 24 * 60 * 60;
         _HTTPCookieAcceptPolicy = NSHTTPCookieAcceptPolicyOnlyFromMainDocumentDomain;
         _HTTPCookieStorage = [[NSHTTPCookieStorage sharedHTTPCookieStorage] retain];
         _HTTPShouldSetCookies = YES;
         _networkServiceType = NSURLNetworkServiceTypeDefault;
         _requestCachePolicy = NSURLRequestUseProtocolCachePolicy;
         _HTTPMaximumConnectionsPerHost = 6;
-        //_URLCredentialStorage = [[NSURLCredentialStorage sharedCredentialStorage] retain];
-    }
-
-    return self;
-}
-
-- (instancetype)_initTLSAndCache {
-    if (self = [self _initWithSharedDefaults]) {
         _TLSMaximumSupportedProtocol = kTLSProtocol12;
         _TLSMinimumSupportedProtocol = kTLSProtocol1;
-        _URLCache = [[NSURLCache sharedURLCache] retain];
+        //_URLCredentialStorage = [[NSURLCredentialStorage sharedCredentialStorage] retain];
     }
-    return self;
-}
 
-/**
- @Status Interoperable
-*/
-- (instancetype)init {
-    if (self = [self _initWithSharedDefaults]) {
-        _TLSMaximumSupportedProtocol = kTLSProtocol12;
-        _TLSMinimumSupportedProtocol = kSSLProtocol3;
-    }
     return self;
 }
 
 /**
  @Status Interoperable
 */
 - (instancetype)initEphemeralSession {
-    self = [self _initTLSAndCache];
+    if (self = [self init]) {
+        _URLCache = [[NSURLCache sharedURLCache] retain];
+    }
     return self;
 }
 
 /**
  @Status Interoperable
 */
 - (instancetype)initBackgroundSession:(NSString*)identifier {
-    if (self = [self _initTLSAndCache]) {
+    if (self = [self init]) {
         _identifier = identifier;
```

Comment:
and released.

---
In reply to: [73748509](https://github.com/Microsoft/WinObjC/pull/828#discussion_r73748509) [](ancestors = 73748509)